### PR TITLE
Docs: Add note about RC deps

### PR DESCRIPTION
**Pull request by delbaoliveira**
Add helper notes for those struggling to upgrade to React 19 RC when using Next.js RC. [NDX-118](https://linear.app/vercel/issue/NDX-118/update-note-on-react-rc-versions-when-upgrading-to-nextjs-15-rc)

### CODE REVIEW

File: docs/02-app/01-building-your-application/11-upgrading/02-version-15.mdx [hunk 1]
```diff
@@ -25,7 +25,10 @@ pnpm up next@rc react@rc react-dom@rc eslint-config-next@rc
 bun add next@rc react@rc react-dom@rc eslint-config-next@rc
 ```
 
-> **Good to know:** If you are using TypeScript, ensure you also upgrade `@types/react` and `@types/react-dom` to their latest versions.
+> **Good to know:**
+>
+> - If you see a peer dependencies warning, you may need to update `react` and `react-dom` to the suggested versions, or you use the `--force` or `--legacy-peer-deps` flag to ignore the warning. This won't be necessary once both Next.js 15 and React 19 are stable.
+> - If you are using TypeScript, ensure you also upgrade `@types/react` and `@types/react-dom` to their latest versions.
```
Comment (arlyon) [inline docs/02-app/01-building-your-application/11-upgrading/02-version-15.mdx:31]: The release blog post from react recommended types-react and types-react-dom, is that still the case?

- https://react.dev/blog/2024/04/25/react-19-upgrade-guide#installing
- https://www.npmjs.com/package/types-react/v/19.0.0-rc.1
- https://www.npmjs.com/package/types-react-dom/v/19.0.0-rc.1

Comment (delbaoliveira) [reply docs/02-app/01-building-your-application/11-upgrading/02-version-15.mdx]: Linked to the React RC upgrade guide.

File: docs/02-app/01-building-your-application/11-upgrading/02-version-15.mdx [hunk 2]
```diff
@@ -25,7 +25,10 @@ pnpm up next@rc react@rc react-dom@rc eslint-config-next@rc
 bun add next@rc react@rc react-dom@rc eslint-config-next@rc
 ```
 
-> **Good to know:** If you are using TypeScript, ensure you also upgrade `@types/react` and `@types/react-dom` to their latest versions.
+> **Good to know:**
+>
+> - If you see a peer dependencies warning, you may need to update `react` and `react-dom` to the suggested versions, or you use the `--force` or `--legacy-peer-deps` flag to ignore the warning. This won't be necessary once both Next.js 15 and React 19 are stable.
+> - If you are using TypeScript, you'll need to temporarily override the React types, see the [React 19 RC upgrade guide](https://react.dev/blog/2024/04/25/react-19-upgrade-guide#installing) for more information.
```
Comment (timeyoutakeit) [inline docs/02-app/01-building-your-application/11-upgrading/02-version-15.mdx:31]:
```suggestion
> - If you are using TypeScript, you'll need to temporarily override the React types. See the [React 19 RC upgrade guide](https://react.dev/blog/2024/04/25/react-19-upgrade-guide#installing) for more information.
```

File: docs/02-app/01-building-your-application/11-upgrading/02-version-15.mdx [hunk 3]
```diff
@@ -25,7 +25,10 @@ pnpm up next@rc react@rc react-dom@rc eslint-config-next@rc
 bun add next@rc react@rc react-dom@rc eslint-config-next@rc
 ```
 
-> **Good to know:** If you are using TypeScript, ensure you also upgrade `@types/react` and `@types/react-dom` to their latest versions.
+> **Good to know:**
+>
+> - If you see a peer dependencies warning, you may need to update `react` and `react-dom` to the suggested versions, or you use the `--force` or `--legacy-peer-deps` flag to ignore the warning. This won't be necessary once both Next.js 15 and React 19 are stable.
```
Comment (timeyoutakeit) [inline docs/02-app/01-building-your-application/11-upgrading/02-version-15.mdx:30]: Suggestion: Is there value in including the wording of the warning to make this note more easily searchable?

**Review by timeyoutakeit — APPROVED** [review]

File: docs/02-app/01-building-your-application/11-upgrading/02-version-15.mdx (hunk 3 above)
Comment (delbaoliveira) [reply docs/02-app/01-building-your-application/11-upgrading/02-version-15.mdx:30]: The React version may change so I opted to not do it.

**Pull request merged**